src/ast/ast_parse.cpp

```diff
@@ -461,6 +506,190 @@ namespace das {
+            serializer_write->metadata = saveRequireMetadata(req);
+    };
+
+    ProgramPtr compileDaScriptSerialize ( const string & fileName,
+                                const FileAccessPtr & access,
+                                TextWriter & logs,
+                                ModuleGroup & libGroup,
+                                CodeOfPolicies policies ) {
+        ReuseCacheGuard rcg;
+        bool exportAll = policies.export_all;
+        auto time0 = ref_time_ticks();
+        totParse = 0;
+        totInfer = 0;
+        totOpt = 0;
+        totM = 0;
+        daScriptEnvironment::bound->macroTimeTicks = 0;
+        vector<ModuleInfo> req;
+        vector<string> missing, circular, notAllowed;
+        das_set<string> dependencies;
+        uint64_t preqT = 0;
+        bool gotReqs = getPrerequisits(fileName, access, req, missing, circular, notAllowed,
+                                        dependencies, libGroup, nullptr, 1, false);
+        DAS_VERIFYF(gotReqs, "expected to get prerequisites for serialization");
+        preqT = get_time_usec(time0);
+        updateSerializationMetadata(req);
+        if ( policies.debugger )
+            addExtraDependency("debug", policies.debug_module, missing, circular, notAllowed, req, dependencies, access, libGroup, policies);
+        if ( policies.profiler )
+            addExtraDependency("profiler", policies.profile_module, missing, circular, notAllowed, req, dependencies, access, libGroup, policies);
+        for ( auto & mod : req ) {
+            if ( libGroup.findModule(mod.moduleName) ) { continue; }
+            auto program = parseDaScript(mod.fileName, access, logs, libGroup, true, true, policies);
+            policies.threadlock_context |= program->options.getBoolOption("threadlock_context",false);
+            if ( program->failed() ) { return program; }
+            if ( policies.fail_on_lack_of_aot_export && !aotModuleHasName(program, mod) ) { return program; }
+            if ( program->thisModule->name.empty() ) {
+                program->thisModule->name = mod.moduleName;
+                program->thisModule->wasParsedNameless = true;
+            }
+            if ( program->promoteToBuiltin ) {
+                if ( canShareModule(program) ) {
+                    program->thisModule->promoteToBuiltin(access);
+                } else {
+                    return program;
+                }
+            }
+            addNewModules(libGroup, program);
+        }
+        auto res = parseDaScript(fileName, access, logs, libGroup, exportAll, false, policies);
+        policies.threadlock_context |= res->options.getBoolOption("threadlock_context",false);
+        if ( !res->failed() ) {
+            if ( res->options.getBoolOption("log_symbol_use") ) {
+                res->markSymbolUse(false, false, false, nullptr, &logs);
+            }
+        }
+        if ( policies.aot_module && (res->promoteToBuiltin || res->thisModule->isModule || exportAll) ) {
```

Comment:
thats a lot of copy-paste from the function bellow. is it a good idea to split so much?

Reply:
The idea was to split features from refactoring. Now that I think about it, makes no sense. Fixed